src/all_types.hpp

```diff
@@ -2235,6 +2244,8 @@ struct CodeGen {
     bool enable_dump_analysis;
     bool enable_doc_generation;
     bool disable_bin_generation;
+    LLVMCodeModel code_model;
+    CodeModel want_code_model;
```

Comment:
I think we only need 1 field here:
```c++
CodeModel code_model;
```
`codegen_create` can default this to `CodeModelDefault`. main.cpp can override it.
The code that generates builtin.zig can switch on the CodeModel (not the LLVM one).
`detect_code_model` can be renamed to `to_llvm_code_model`, and only called once, for `createTargetMachine`.

The `want_*` mechanisms don't apply to this parameter, since it has a default and there is no logic in the "detect" phase.